crates/integrations/datafusion/src/table/mod.rs

```diff
@@ -130,10 +131,14 @@ impl TableProvider for IcebergTableProvider {
         filters: &[Expr],
         _limit: Option<usize>,
     ) -> DFResult<Arc<dyn ExecutionPlan>> {
+        let statistics = compute_statistics(&self.table, self.snapshot_id)
+            .await
+            .unwrap_or(Statistics::new_unknown(self.schema.as_ref()));
```

Comment:
Arguably this should be computed when instantiating the `IcebergTableProvider` (consequently `IcebergTableProvider::new` would become async).

That way not only could `TableProvider::statistics` be exposed, we'd save a const perf penalty during planning for reading the manifests.